DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/ViewTreeSnapshotBuilder.swift

```diff
@@ -17,8 +17,8 @@ internal struct ViewTreeSnapshotBuilder {
     let idsGenerator: NodeIDGenerator
     /// Provides base64 image data with a built in caching mechanism.
     let imageDataProvider: ImageDataProviding
-    /// Available text obfuscators to use by different recorders in response to current privacy mode.
-    let textObfuscators: TextObfuscators
+    /// Text obfuscation strategies for different text types.
+    let textObfuscation: TextObfuscation
```

Comment:
`TextObfuscation` is not an abstraction so we don't get much value of injecting it, using an `SessionReplayPrivacy` extension as mentioned would simplify things.

Reply:
Agree - although as explained, I was anticipating adding cache later. That way, the `VTSBuilder` would orchestrate it. Anyway, this was revamped according to your proposal ✅